### PR TITLE
docs: clean up world api.md a bit

### PR DESCRIPTION
**Pull request by el-u**
## What is this fixing or adding?

This fixes some mistakes and inconsistencies in "world api.md".

### CODE REVIEW

File: docs/world api.md [hunk 1]
```diff
@@ -135,10 +137,10 @@ same ID. Name must not be numeric (has to contain at least 1 letter or symbol).
...
 
 Other classifications include
-* filler: a regular item or trash item
-* useful: generally quite useful, but not required for anything logical
-* trap: negative impact on the player
-* skip_balancing: add to progression to skip balancing; e.g. currency or tokens
+* `filler`: a regular item or trash item
+* `useful`: generally quite useful, but not required for anything logical
+* `trap`: negative impact on the player
+* `skip_balancing`: add to `progression` to skip balancing; e.g. currency or tokens
```
Comment (kindasneaki) [inline docs/world api.md:143]:
```suggestion
* `skip_balancing`: adds a `progression` item that skips progression balancing; e.g. currency or tokens
```
I feel like the wording here is confusing

Comment (kindasneaki) [reply docs/world api.md:143]: Oh I misunderstood what skip_balancing is then, but it should have `progression_skip_balancing` as well as `skip_balancing`.

Comment (alwaysintreble) [reply docs/world api.md:143]: `progression_skip_balancing` technically isn't a flag; it's a combination of the two flags that's used as shorthand. It still exists and is kind of "expected" use, so probably a good idea to also list it here.

**Review by black-sliver — APPROVED** [review]
I didn't realize there were that many multiworld renames missed.

As for [...] vs. <...> i think i went for [] to avoid accidentally creating html tags, but it looks like it renders correctly, so the change is good.

**Pull request merged**